components/camel-salesforce/camel-salesforce-component/src/main/java/org/apache/camel/component/salesforce/api/dto/composite/ReferenceId.java

```diff
@@ -0,0 +1,89 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component.salesforce.api.dto.composite;
+
+import java.io.Serializable;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.thoughtworks.xstream.annotations.XStreamAlias;
+import com.thoughtworks.xstream.annotations.XStreamImplicit;
+
+import org.apache.camel.component.salesforce.api.dto.RestError;
+
+@XStreamAlias("results")
+public final class ReferenceId implements Serializable {
+    private static final long serialVersionUID = 1L;
+
+    @XStreamImplicit
+    private final List<RestError> errors;
+
+    private final String id;
+
+    private final String referenceId;
+
+    @JsonCreator
+    ReferenceId(@JsonProperty("referenceId") final String referenceId, @JsonProperty("id") final String id,
+            @JsonProperty("errors") final List<RestError> errors) {
+        this.referenceId = referenceId;
+        this.id = id;
+        this.errors = Optional.ofNullable(errors).orElse(Collections.emptyList());
+    }
+
+    public List<RestError> getErrors() {
+        return Optional.ofNullable(errors).orElse(Collections.emptyList());
+    }
+
+    public String getId() {
+        return id;
+    }
+
+    public String getReferenceId() {
+        return referenceId;
+    }
+
+    @Override
+    public boolean equals(final Object obj) {
+        if (obj == this) {
+            return true;
+        }
+
+        if (!(obj instanceof ReferenceId)) {
+            return false;
+        }
+
+        final ReferenceId other = (ReferenceId) obj;
+
+        return Objects.equals(id, other.id) && Objects.equals(referenceId, other.referenceId)
+            && Objects.equals(getErrors(), other.getErrors());
+    }
+
+    @Override
+    public int hashCode() {
+        return Arrays.hashCode(new Object[] {id, referenceId, errors});
+    }
+
+    @Override
+    public String toString() {
```

Comment:
could also be a bit too verbose, made it so that only number of errors if any is returned in 76953eecff5b011961598cb8f44dcb522f8245f9